aten/src/ATen/native/native_functions.yaml

```diff
@@ -14620,11 +14620,16 @@
     CUDA: _scaled_dot_product_efficient_attention_backward_cuda
   tags: nondeterministic_seeded
 
-- func: _scaled_dot_product_cudnn_attention(Tensor query, Tensor key, Tensor value, float dropout_p=0.0, bool is_causal=False, bool return_debug_mask=False, *, float? scale=None) -> (Tensor output, Tensor logsumexp, Tensor philox_seed, Tensor philox_offset)
+- func: _scaled_dot_product_cudnn_attention(Tensor query, Tensor key, Tensor value, float dropout_p=0.0, bool is_causal=False, bool return_debug_mask=False, *, float? scale=None) -> (Tensor output, Tensor logsumexp, Tensor cum_seq_q, Tensor cum_seq_k, SymInt max_q, SymInt max_k, Tensor philox_seed, Tensor philox_offset, Tensor debug_attn_mask)
```

Comment:
Why is this a special function? @jbschlosser and @andrewor14 have spent litteral months removing these cudnn variants for conv and batchnorm. I really don't think we should be doing the same again with sdpa...

Reply:
I think in this case it's because cuDNN is just a backend in the same way that e.g., flash, mem-efficient, and math are separate backends because the rules for dispatching between them aren't expected to overlap (e.g., cuDNN SDPA isn't expected to exactly cover the support matrix of flash or mem-efficient attention)

Reply:
Me and @albanD had a long discussion about this on Friday. Historically there has been a similar pattern for aten.op bloat followed by the need to unify a large number of backend behind composite explicit ops. The two exemplars of this are torch's cuDNN op, and batch norm with which consolidation is still ongoing by @andrewor14. 

The conclusion was that for now we are okay with paying the backend tech debt (added ops and added coverage surface) for the sake of velocity.